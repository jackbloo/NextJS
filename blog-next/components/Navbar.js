import styles from '../styles/Home.module.css'
import Link from 'next/link';


function Nav({data, title,id}){
return(
    <div className={styles.nav} style={{fontWeight: data ? 'bold' : 'normal', textDecoration: data ? 'underline' : 'none'}}>
    <Link href={`/${id}`}><a>{title}</a></Link>
    </div>
)

}


export default function Navbar({title}){
    const data = [{id:'', title:'Home'},{id:'about', title:'About us'}]
    return(
        <div className={styles.navbar} data-testid="navbar-container">
        <div className={styles.rightNavbar}>
        <Link href="/"><a>Digitsense Blog</a></Link>
        </div>
        <div className={styles.leftNavbar}>
            {data.map((el,i) => <Nav data={title === el.id} title={el.title} id={el.id} key={`${el.id} - ${i}`}/>)}
            <Nav/>
        </div>
      </div>
    )
}