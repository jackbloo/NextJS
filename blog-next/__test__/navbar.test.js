import { render, screen } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import Navbar from '../components/Navbar'

describe('Unit Testing of Navbar', () => {
  it('Should renders navbar container', async () => {
      await act(async () =>{
        await render(<Navbar title="about"/>)
      })

    const heading = screen.getByTestId('navbar-container')
    expect(heading).toMatchSnapshot();
  })
})