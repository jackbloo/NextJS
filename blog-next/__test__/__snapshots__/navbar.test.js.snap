// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing of Navbar Should renders navbar container 1`] = `
<div
  class="navbar"
  data-testid="navbar-container"
>
  <div
    class="rightNavbar"
  >
    <a
      href="/"
    >
      Digitsense Blog
    </a>
  </div>
  <div
    class="leftNavbar"
  >
    <div
      class="nav"
      style="font-weight: normal; text-decoration: none;"
    >
      <a
        href="/"
      >
        Home
      </a>
    </div>
    <div
      class="nav"
      style="font-weight: bold; text-decoration: underline;"
    >
      <a
        href="/about"
      >
        About us
      </a>
    </div>
    <div
      class="nav"
      style="font-weight: normal; text-decoration: none;"
    >
      <a
        href="/undefined"
      />
    </div>
  </div>
</div>
`;
